api/v1alpha1/spire_server_config_types.go

```diff
@@ -165,10 +171,182 @@ type CASubject struct {
+
+	// issuerKind is the kind of the issuer to reference in CertificateRequests. Defaults to "Issuer" if empty.
+	// +kubebuilder:validation:Optional
+	IssuerKind string `json:"issuerKind,omitempty"`
+
+	// issuerGroup is the group of the issuer to reference in CertificateRequests. Defaults to "cert-manager.io" if empty.
+	// +kubebuilder:validation:Optional
+	IssuerGroup string `json:"issuerGroup,omitempty"`
+
+	// namespace in which to create CertificateRequests for signing.
+	// +kubebuilder:validation:Optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// kubeConfigSecretName is the name of the Kubernetes Secret that stores the kubeconfig
+	// used to connect to the Kubernetes cluster where cert-manager is running.
+	// If empty, in-cluster configuration will be used.
+	// +kubebuilder:validation:Optional
+	KubeConfigSecretName string `json:"kubeConfigSecretName,omitempty"`
+}
+
+// UpstreamAuthorityVault contains the configuration required to use
+// HashiCorp Vault as the upstream authority for SPIRE.
+// It supports multiple authentication mechanisms including Token, Cert, AppRole, and Kubernetes.
+type UpstreamAuthorityVault struct {
+	// VaultAddress is the URL of the Vault server (e.g., https://vault.example.com:8443).
+	// +kubebuilder:validation:Required
+	VaultAddress string `json:"vaultAddress,omitempty"`
+
+	// Namespace is the Vault Enterprise namespace (optional).
+	// +kubebuilder:validation:Optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// PkiMountPoint is the mount point where the PKI secrets engine is enabled (e.g., "pki").
+	// +kubebuilder:validation:Required
+	PkiMountPoint string `json:"pkiMountPoint,omitempty"`
+
+	// CaCertSecret is the name of the Kubernetes secret that contains the CA certificate (PEM format).
+	// +kubebuilder:validation:Required
+	CaCertSecret string `json:"caCertSecret,omitempty"`
+
+	// TokenAuth configures Vault token-based authentication.
+	// +kubebuilder:validation:Optional
+	TokenAuth *TokenAuth `json:"tokenAuth,omitempty,omitempty"`
+
+	// CertAuth configures Vault client certificate authentication.
+	// +kubebuilder:validation:Optional
+	CertAuth *CertAuth `json:"certAuth,omitempty"`
+
+	// AppRoleAuth configures Vault AppRole authentication.
+	// +kubebuilder:validation:Optional
+	AppRoleAuth *AppRoleAuth `json:"appRoleAuth,omitempty"`
+
+	// K8sAuth configures Vault Kubernetes authentication.
+	// +kubebuilder:validation:Optional
+	K8sAuth *K8sAuth `json:"k8sAuth,omitempty"`
+}
+
+// TokenAuth configures the Vault token authentication method.
+// The token is used as a bearer token in the "X-Vault-Token" header.
+type TokenAuth struct {
+	// Token is the Vault token string.
+	// +kubebuilder:validation:Required
+	Token string `json:"token,omitempty"`
+}
+
+// CertAuth configures the Vault client certificate authentication method.
+type CertAuth struct {
+	// CertAuthMountPoint is the mount point where the TLS certificate auth method is enabled.
```

Comment:
There are contradicting validation mentioned
openAPISchema mandates the field while, go json marshal will ignore this field if it is empty